CMakeLists.txt

```diff
@@ -20,70 +20,61 @@ if( CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
     ecbuild_add_fortran_flags("-fallow-argument-mismatch")
   endif()
+  if( CMAKE_Fortran_COMPILER_VERSION VERSION_LESS "11.0")
+    set( ENABLE_ACC OFF )
+  endif()
 endif()
 
 ### OpenACC
 if( NOT DEFINED ENABLE_ACC OR ENABLE_ACC )
-    find_package( OpenACC COMPONENTS Fortran C )
-    # Ensure targets are available
-    if( OpenACC_Fortran_FOUND AND NOT TARGET OpenACC::OpenACC_Fortran )
-        ecbuild_critical( "OpenACC::OpenACC_Fortran should have been defined as a CMake target" )
-    endif()
-    if( OpenACC_C_FOUND AND NOT TARGET OpenACC::OpenACC_C )
-        ecbuild_critical( "OpenACC::OpenACC_C should have been defined as a CMake target" )
+    # Incredibly inconvenient: FindOpenACC does _not_ set OpenACC_FOUND, only
+    # the language-specific components OpenACC_Fortran_FOUND and OpenACC_C_FOUND.
+    # This means, even internally CMake considers OpenACC as not found.
+    # (See eg get_property(... GLOBAL PROPERTY PACKAGES_NOT_FOUND))
+    # Therefore, we search for OpenACC, set OpenACC_FOUND ourselves according to
+    # the result, and then, trigger a second find_package via ecbuild_add_option.
+    # This then conveniently takes the previously set OpenACC_FOUND into account
+    # and rectifies CMake's internal bookkeeping in the process.
+    find_package( OpenACC )
+    if( OpenACC_Fortran_FOUND AND OpenACC_C_FOUND )
+        set( OpenACC_FOUND ON )
     endif()
 endif()
 ecbuild_add_option( FEATURE ACC
     DESCRIPTION "OpenACC" DEFAULT ON
-    CONDITION OpenACC_C_FOUND OR OpenACC_Fortran_FOUND )
+    REQUIRED_PACKAGES "OpenACC" )
 
 ### OpenMP
-if( NOT DEFINED ENABLE_OMP OR ENABLE_OMP )
-    find_package( OpenMP COMPONENTS Fortran C )
-    # Ensure targets are available
-    if( OpenMP_Fortran_FOUND AND NOT TARGET OpenMP::OpenMP_Fortran )
-        ecbuild_critical( "OpenMP::OpenMP_Fortran should have been defined as a CMake target" )
-    endif()
-    if( OpenMP_C_FOUND AND NOT TARGET OpenMP::OpenMP_C )
-        ecbuild_critical( "OpenMP::OpenMP_C should have been defined as a CMake target" )
-    endif()
-endif()
 ecbuild_add_option( FEATURE OMP
     DESCRIPTION "OpenMP" DEFAULT ON
-    CONDITION OpenMP_C_FOUND AND OpenMP_Fortran_FOUND )
+    REQUIRED_PACKAGES "OpenMP COMPONENTS Fortran C" )
+include(features/OMP)
 
 ### MPI
-if( NOT DEFINED ENABLE_MPI OR ENABLE_MPI )
-    ecbuild_find_mpi( COMPONENTS Fortran )
-endif()
-if( MPI_Fortran_FOUND AND ENABLE_MPI )
-    # Ensure targets are available
-    if( MPI_Fortran_FOUND AND NOT TARGET MPI::MPI_Fortran )
-        ecbuild_critical( "MPI::MPI_Fortran should have been defined as a CMake target" )
-    endif()
-    ecbuild_add_option( FEATURE MPI
-        DESCRIPTION "MPI" DEFAULT OFF
-        CONDITION MPI_Fortran_FOUND )
+ecbuild_add_option( FEATURE MPI
+    DESCRIPTION "MPI" DEFAULT OFF
+    REQUIRED_PACKAGES "MPI COMPONENTS Fortran" )
+if( HAVE_MPI )
+    ecbuild_enable_mpi( COMPONENTS Fortran )
     list(APPEND CLOUDSC_DEFINITIONS HAVE_MPI ${MPI_DEFINITIONS})
```

Comment:
I would not use `ecbuild_enable_mpi` and also not append `${MPI_DEFINITIONS}`
Rather the `MPI::MPI_Fortran` target should be linked where appropriate, if not already done.

Reply:
In fact, we had the same redundant construction for HDF5_Fortran_DEFINITIONS